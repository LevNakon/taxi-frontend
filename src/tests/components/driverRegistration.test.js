import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import DriverRegistration from '../../components/driverRegistration';

describe('DriverRegistration --- test render correctly component', () => {
    const initialState = {
        carState:{
            message: '',
            success: true,
            car: {
                brand: 'test brand',
                model: 'test model',
                year: 2000,
                run: 20000,
            }
        },
        driverState: {
            message: '',
            success: true,
            driver: {
                experience: 0,
                condition: 'COMFORT',
            }
        }
    };
    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<DriverRegistration store={store} />);
    });

    it('render correctly component DriverRegistration', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('check props matches with initialState', () => {
        expect(wrapper.dive().prop('successDriver')).toEqual(initialState.driverState.success);
        expect(wrapper.dive().prop('messageDriver')).toEqual(initialState.driverState.message);
        expect(wrapper.dive().prop('driver')).toEqual(initialState.driverState.driver);
        expect(wrapper.dive().prop('successCar')).toEqual(initialState.carState.success);
        expect(wrapper.dive().prop('messageCar')).toEqual(initialState.carState.message);
        expect(wrapper.dive().prop('car')).toEqual(initialState.carState.car);
    });
});