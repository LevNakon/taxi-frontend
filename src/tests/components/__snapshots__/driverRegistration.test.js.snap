// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriverRegistration --- test render correctly component render correctly component DriverRegistration 1`] = `
<ContextProvider
  value={null}
>
  <withRouter(DriverRegistration)
    car={
      Object {
        "brand": "test brand",
        "model": "test model",
        "run": 20000,
        "year": 2000,
      }
    }
    carCreateNull={[Function]}
    driver={
      Object {
        "condition": "COMFORT",
        "experience": 0,
      }
    }
    driverCarWatcher={[Function]}
    driverCreateNull={[Function]}
    messageCar=""
    messageDriver=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    successCar={true}
    successDriver={true}
  />
</ContextProvider>
`;
